### PR TITLE
Added cleanup to build and publish actions

### PR DESCRIPTION
**Pull request by chef-delivery**
This Pull Request was opened by Chef Automate user Davida Marion

Signed-off-by: Davida Marion <dmarion@chef.io>

Ready to merge? View this change in Chef Automate and click the Approve button: https://delivery.chef.co/e/chef/#/organizations/Delivery-Build-Cookbooks/projects/habitat-build/changes/457f18a8-8b19-41cd-80b0-f06fabad9422

### CODE REVIEW

File: resources/hab_build.rb [hunk 1]
```diff
@@ -67,6 +68,12 @@ def last_build_env
     retries new_resource.retries
     live_stream new_resource.live_stream
   end
+
+  execute 'remove-studio' do
+    command "sudo -E #{hab_binary} studio rm #{hab_studio_path}"
+    live_stream new_resource.live_stream
+    only_if { new_resource.cleanup }
```
Comment (jtimberman) [inline resources/hab_build.rb]: Generally it is preferred style in custom resources to use the property, not `new_resource.property`. Thus, use `cleanup`, not `new_resource.cleanup`.

Ref https://docs.chef.io/custom_resources.html#new-resource-property

File: resources/hab_build.rb [hunk 2]
```diff
@@ -13,6 +13,7 @@
 property :home_dir, String
 property :auth_token, String
 property :live_stream, [TrueClass, FalseClass], default: true
+property :cleanup, [TrueClass, FalseClass], default: false
```
Comment (jtimberman) [inline resources/hab_build.rb]: I learned recently that for these kinds of properties, the allowed values should be `[true,  false]`. Please do that for `cleanup`, and while we're in this file, `live_stream` too 💃 .

Ref: https://github.com/chef-cookbooks/habitat/pull/31#discussion_r112523204

File: resources/hab_build.rb [hunk 3]
```diff
@@ -84,6 +91,12 @@ def last_build_env
     live_stream new_resource.live_stream
     sensitive true
   end
+
+  file 'artifact' do
+    path lazy { "#{hab_studio_path}/src/results/#{artifact}" }
+    action :delete
+    only_if { new_resource.cleanup }
```
Comment (jtimberman) [inline resources/hab_build.rb]: as above, use `cleanup`

File: resources/hab_build.rb [hunk 4]
```diff
@@ -84,6 +91,12 @@ def last_build_env
     live_stream new_resource.live_stream
     sensitive true
   end
+
+  file 'artifact' do
```
Comment (jtimberman) [inline resources/hab_build.rb]: `file 'remove-artifact'` or something clear as to what the artifact resource here is.

File: spec/unit/resources/hab_build_spec.rb [hunk 5]
```diff
@@ -58,6 +82,20 @@
...
+    it 'does not delete the artifact' do
+      expect(chef_run).not_to delete_file('artifact')
+    end
+  end
+
+  context 'when cleanup is set to true' do
+    let(:cleanup) { true }
+
+    it 'deletes the artifact' do
+      expect(chef_run).to delete_file('artifact')
```
Comment (jtimberman) [inline spec/unit/resources/hab_build_spec.rb]: Remember to change this to match the file resource change as mentioned above.

File: resources/hab_build.rb (hunk 2 above)
Comment (phiggins) [inline resources/hab_build.rb]: It doesn't say how the values are compared to the type (ie, `is_a?` vs `===`), but it shows `true` and `false` as example types in the docs: https://docs.chef.io/custom_resources.html#property

File: spec/unit/resources/hab_build_spec.rb [hunk 6]
```diff
@@ -7,12 +7,16 @@
       allow(File).to receive(:read).and_call_original
     end
```
Comment (phiggins) [inline spec/unit/resources/hab_build_spec.rb]: Not related to your PR, but I'm pretty sure you can remove the entire before block on lines 6-8.

File: spec/unit/resources/hab_build_spec.rb (hunk 5 above)
Comment (phiggins) [inline spec/unit/resources/hab_build_spec.rb]: Shouldn't this test fail if one place is updated and the other isn't?

**Pull request closed without merging**